service-api/module/Application/src/Controller/YotiController.php

```diff
@@ -82,4 +81,52 @@ public function getSessionStatusAction(): JsonModel
 
         return new JsonModel($data);
     }
+
+    /**
+     * @return JsonModel
+     */
+    public function notificationAction(): JsonModel
+    {
+        $data = json_decode($this->getRequest()->getContent(), true);
+        if (! isset($data['session_id'], $data['topic'])) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(new Problem('Missing required parameters'));
+        }
+
+        if (! in_array($data['topic'], ['first_branch_visit', 'session_completion'])) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(new Problem('Invalid type'));
+        }
+
+        $sessionId = filter_var($data['session_id'], FILTER_SANITIZE_SPECIAL_CHARS);
+
+        $caseData = $this->dataQuery->queryByYotiSessionId($sessionId);
```

Comment:
This wasn't in the ticket because we ended up doing things in a different order, but we need to authenticate the auth token too by checking that the `Authorization` header matches `$caseData->counterService->notificationsAuthToken`.

NB: the Authorization header will be in the format `Bearer <TOKEN>`, so you need to strip the Bearer bit off first

Reply:
Doing this on the follow on ticket ID-170 right now :)

Reply:
🤦🏽 Missed that one, thanks